omc3/scripts/kmod_lumi_imbalance.py

```diff
@@ -0,0 +1,237 @@
+            except KeyError as e:
+                msg = (
+                    f"Dataframe '{df}' does not contain a '{BEAM}' column."
+                    "You need to run the `kmod_average` script on data for both beams "
+                    "before you can calulate the luminosity imbalance."
+                )
+                raise KeyError(msg) from e
+
+        if BEAM in df.columns:
+            df = df.set_index(BEAM)
+
+        for beam in (1, 2):
+            if beam not in df.index:
+                msg = (
+                    f"Dataframe '{df}' does not seem to contain data per beam."
+                    "You need to run the `kmod_average` script on data for both beams "
+                    "before you can calulate the luminosity imbalance."
+                )
+                raise KeyError(msg)
+
+        dfs[ip] = df
+    
+    if len(dfs) != 2:
+        msg = (
+            "Lumi inbalance can only be calculated for exactly two IPs, "
+            f"but instead {len(dfs)} were given."
+        )
+        raise ValueError(msg)
+
+    return dfs
+
+
+def get_lumi_imbalance_df(**kwargs) -> tfs.TfsDataFrame:
+    """
+    Calculate the effective beta stars and the luminosity imbalance from the input dataframes.
+
+    Args:
+        ipA (tfs.TfsDataFrame): ar`TfsDataFrame` with the averaged results from a kmod analysis, for IP_A.
+        ipB (tfs.TfsDataFrame): a `TfsDataFrame` with the averaged results from a kmod analysis, for IP_B.
+        (Actually, any name that ends with an integer is fine.)
+    
+    Returns:
+        tfs.TfsDataFrame with effective beta stars per IP and the luminosity imbalance added to the header.
+    """
+    df_effective_betas = tfs.TfsDataFrame()
+    for ip, df in kwargs.items():
+        df_effective_betas.loc[ip.upper(), [f'{BETASTAR}', f'{ERR}{BETASTAR}']] = get_effective_beta_star_w_err(df)
+    
+    ip_a, ip_b = df_effective_betas.index
+    lumi_imb, lumi_imb_err = get_imbalance_w_err(
+        *tuple(df_effective_betas.loc[ip_a, :]), 
+        *tuple(df_effective_betas.loc[ip_b, :])
+    )
+    
+    df_effective_betas.headers[f'{LUMINOSITY}{IMBALACE}'] = lumi_imb
+    df_effective_betas.headers[f'{ERR}{LUMINOSITY}{IMBALACE}'] = lumi_imb_err
+    return df_effective_betas
+
+
+def get_imbalance_w_err(ipA_beta: float, ipA_beta_err: float, ipB_beta: float, ipB_beta_err: float) -> tuple[float, float]:
+    """
+    Calculate the luminosity imbalance IP_A / IP_B  and its error.
+    """
+    result = ipB_beta / ipA_beta  # inverse due to beta in the denominator for lumi
+    err = result * np.sqrt((ipB_beta_err/ipB_beta)**2 + (ipA_beta_err/ipA_beta)**2)
+    return result, err
+
+
+def get_effective_beta_star_w_err(df_ip: tfs.TfsDataFrame) -> tuple[float]:
+    """ Calculates the effective beta*, 
+    i.e. the denominator of the luminosity (e.g. Eq(17): https://cds.cern.ch/record/941318/files/p361.pdf)
+    without any constants, as we only need it for the ratio anyway.  
+    """
```

Comment:
Same here, we should write down the important caveat as above: Eq 17 in this ref is a pretty simplified calculation and later on we might have to use a more accurate one.